neuralhydrology/datasetzoo/camelsind.py

```diff
@@ -0,0 +1,410 @@
+
+    print("The following columns are now available in each per-basin CSV file.")
+    print("Use these names in your NeuralHydrology config.yml file:")
+    print(f"  > For 'dynamic_inputs': {dynamic_inputs}")
+    if targets:
+        print(f"  > For 'target_variables': {targets}")
+    else:
+        print("  > Warning: No standard target variable like 'streamflow' was found.")
+
+    print(f"\nAll per-basin time series files have been stored in: {timeseries_output_dir}")
+
+
+def _process_and_merge_attributes(base_path: Path, attributes_output_file: Path):
+    """Merges all static attribute CSVs from a directory into a single file.
+
+    This function scans a specified `attributes_csv` subdirectory for all files
+    ending with '.csv'. It reads each file and iteratively merges them into a
+    single pandas DataFrame using an outer join on the 'gauge_id' column.
+    Columns that are entirely empty (all NaN) after the merge are removed.
+    The final, consolidated attributes are saved to a new CSV file.
+
+    Parameters
+    ----------
+    base_path : Path
+        The root directory path of the dataset. This function expects it to
+        contain an `attributes_csv` subdirectory.
+    attributes_output_file : Path
+        The full path, including filename, where the final merged attributes
+        CSV file will be saved.
+
+    Returns
+    -------
+    None
+        This function does not return a value but saves a file to disk and
+        prints progress and summary information to the console.
+    """
+    print("\n--- Step 2: Merging All Attribute Files ---")
+    attributes_folder = base_path / 'attributes_csv'
+
+    if not attributes_folder.is_dir():
+        raise FileNotFoundError(f"Attributes folder not found at: {attributes_folder}")
+
+    csv_files = list(attributes_folder.glob('*.csv'))
+    if not csv_files:
+        print("Warning: No attribute CSV files found. Skipping attribute processing.")
+        return
+
+    dataframes = [pd.read_csv(file) for file in tqdm(csv_files, desc="Reading attribute files")]
+
+    merged_attributes_df = reduce(lambda left, right: pd.merge(left, right, on='gauge_id', how='outer'), dataframes)
+
+    merged_attributes_df = merged_attributes_df.dropna(axis=1, how='all')
+    merged_attributes_df.to_csv(attributes_output_file, index=False)
+    print(f"Successfully merged {len(csv_files)} attribute files.")
+    print(f"Consolidated attributes file saved to: {attributes_output_file}")
+
+
+# --- DOCSTRING UPDATED ---
+def preprocess_camels_ind_dataset(camels_base_path: str, output_dir: str):
+    """Orchestrates the full preprocessing of a CAMELS-IND dataset.
+    
+    The raw dataset can be downloaded from `Zenodo <https://zenodo.org/records/14999580>`_.
+    The output_dir is the processed CAMELS IND directory mentioned in the codes above.
+    This function performs two main tasks:
+    1.  Processes time series data (forcings, streamflow) and saves one CSV
```

Comment:
You might need to have a blank line before this enumeration here, for the list to correctly render.